nifi-nar-bundles/nifi-dbcp-services-bundle/nifi-postgres-dbcp-service/src/main/java/org/apache/nifi/service/PostgreSQLConnectionPool.java

```diff
@@ -0,0 +1,255 @@
+            final boolean kerberosPasswordProvided = !StringUtils.isBlank(context.getProperty(POSTGRES_KERBEROS_PASSWORD).getValue());
+
+            if (kerberosPrincipalProvided && !kerberosPasswordProvided) {
+                results.add(new ValidationResult.Builder()
+                        .subject(POSTGRES_KERBEROS_PASSWORD.getDisplayName())
+                        .valid(false)
+                        .explanation("a password must be provided for the given principal")
+                        .build());
+            }
+
+            if (kerberosPasswordProvided && !kerberosPrincipalProvided) {
+                results.add(new ValidationResult.Builder()
+                        .subject(POSTGRES_KERBEROS_PRINCIPAL.getDisplayName())
+                        .valid(false)
+                        .explanation("a principal must be provided for the given password")
+                        .build());
+            }
+
+            final KerberosCredentialsService kerberosCredentialsService = context.getProperty(POSTGRES_KERBEROS_CREDENTIALS_SERVICE).asControllerService(KerberosCredentialsService.class);
+            final KerberosUserService kerberosUserService = context.getProperty(POSTGRES_KERBEROS_USER_SERVICE).asControllerService(KerberosUserService.class);
+
+            if (kerberosCredentialsService != null && (kerberosPrincipalProvided || kerberosPasswordProvided)) {
+                results.add(new ValidationResult.Builder()
+                        .subject(POSTGRES_KERBEROS_CREDENTIALS_SERVICE.getDisplayName())
+                        .valid(false)
+                        .explanation("kerberos principal/password and kerberos credential service cannot be configured at the same time")
+                        .build());
+            }
+
+            if (kerberosUserService != null && (kerberosPrincipalProvided || kerberosPasswordProvided)) {
+                results.add(new ValidationResult.Builder()
+                        .subject(POSTGRES_KERBEROS_USER_SERVICE.getDisplayName())
+                        .valid(false)
+                        .explanation("kerberos principal/password and kerberos user service cannot be configured at the same time")
+                        .build());
+            }
+
+            if (kerberosUserService != null && kerberosCredentialsService != null) {
+                results.add(new ValidationResult.Builder()
+                        .subject(POSTGRES_KERBEROS_USER_SERVICE.getDisplayName())
+                        .valid(false)
+                        .explanation("kerberos user service and kerberos credential service cannot be configured at the same time")
+                        .build());
+            }
+        }
+        return results;
+    }
+
+    @Override
+    protected Driver getDriver(final String driverName, final String url) {
+        try {
+            Class.forName(org.postgresql.Driver.class.getName());
+            return DriverManager.getDriver(url);

```

Comment:
Instead of using this approach, it should be possible to return an instance of the PostgreSQL driver directly.